src/HDF5.jl

```diff
@@ -1761,49 +1761,49 @@ end
 ### Convenience wrappers ###
 # These supply default values where possible
 # See also the "special handling" section below
-function h5a_write(attr_id::hid_t, mem_type_id::hid_t, str::AbstractString)
+function h5a_write(attr_id, mem_type_id, str::AbstractString)
     strbuf = Base.cconvert(Cstring, str)
     GC.@preserve strbuf begin
         buf = Base.unsafe_convert(Ptr{UInt8}, strbuf)
         h5a_write(attr_id, mem_type_id, buf)
     end
 end
-function h5a_write(attr_id::hid_t, mem_type_id::hid_t, x::T) where {T<:Union{ScalarType,Complex{<:ScalarType}}}
+function h5a_write(attr_id, mem_type_id, x::T) where {T<:Union{ScalarType,Complex{<:ScalarType}}}
     tmp = Ref{T}(x)
     h5a_write(attr_id, mem_type_id, tmp)
 end
-function h5a_write(attr_id::hid_t, memtype_id::hid_t, strs::Array{S}) where {S<:String}
+function h5a_write(attr_id, memtype_id, strs::Array{S}) where {S<:String}
```

Comment:
@jmert  I'm not sure if we need to fix this as well to accept Arrays of abstract strings.

Reply:
It looks like changing it would work with `GenericString`, so I'd vote for yes.:
```julia
julia> using HDF5, Test

julia> hfile = h5open("/tmp/test.h5", "w");

julia> genstrs = GenericString["one", "purple"];

julia> a, dtype = a_create(hfile, "myattr", genstrs)
(HDF5 attribute: myattr, HDF5 datatype: H5T_STRING {
      STRSIZE H5T_VARIABLE;
      STRPAD H5T_STR_NULLTERM;
      CSET H5T_CSET_UTF8;
      CTYPE H5T_C_S1;
   })

julia> HDF5.h5a_write(a, dtype, Ref{Cstring}(genstrs))

julia> flush(hfile)

shell> h5dump -p /tmp/test.h5
HDF5 "/tmp/test.h5" {
GROUP "/" {
   ATTRIBUTE "myattr" {
      DATATYPE  H5T_STRING {
         STRSIZE H5T_VARIABLE;
         STRPAD H5T_STR_NULLTERM;
         CSET H5T_CSET_UTF8;
         CTYPE H5T_C_S1;
      }
      DATASPACE  SIMPLE { ( 2 ) / ( 2 ) }
      DATA {
      (0): "one", "purple"
      }
   }
}
}
```

(I didn't locally modify this branch, so that's the reason for the manual call to `Ref{Cstring}(genstrs)`.)